pulsar-broker/src/test/java/org/apache/pulsar/io/PulsarFunctionE2ETest.java

```diff
@@ -450,6 +452,15 @@ private void testE2EPulsarFunction(String jarFilePathUrl) throws Exception {
         // make sure subscriptions are cleanup
         assertEquals(admin.topics().getStats(sourceTopic).subscriptions.size(), 0);
 
+        // make sure all temp files are deleted
+        File dir = new File(System.getProperty("java.io.tmpdir"));
```

Comment:
Maybe we could have a configurable directory for these download & test operation. Given that /tmp is most commonly mounted on tmpfs which is a ramdisk, we should allow users to configure a different directory that is disk-based (eg: using emptyDir on K8S)

Reply:
@merlimat yup I agree.  That is why I have consolidated the code that creates these temp files.  This PR is already sizable so lets do this in a subsequent PR